docs/spec/annotations.rst

```diff
@@ -354,9 +350,16 @@ Annotating instance and class methods
 -------------------------------------
 
 In most cases the first argument of class and instance methods
-does not need to be annotated, and it is assumed to have the
-type of the containing class for instance methods, and a type object
-type corresponding to the containing class object for class methods.
+(conventionally named ``self`` or ``cls``) does not need to be annotated.
+
+If the argument is not annotated, then for instance methods it is
+assumed to have the type of the containing class or :ref:`Self
+<self>`, and for class methods the type object type corresponding to
+the containing class object or ``type[Self]``. For example, in class
+``A`` the first argument of an instance method has the implicit type
+``A``. In a class method, the precise type of the first argument
+cannot be represented using the available type notation.
```

Comment:
I'm not 100% sure what this last sentence means.

I think `type[Self]` is a fairly precise type - it's a type variable bound above by `type[<current-class>]` but can be narrowed on callsites where we know more.

Reply:
Yeah, that sentence is from PEP 484, it confused me as well. I thought it was maybe talking about the usual thing where args to `cls(...)` are hopeless, but I don't really know

Reply:
Yes, that sentence makes no sense to me either, I think we should remove it.

Reply:
Except for the last sentence, I think it looks reasonable. 

I do think the spec should eventually be updated to mandate that `self` has the implied type `Self` and `cls` has the implied type `type[Self]`. Many important use cases involving the `Self` type break if this isn't the case. In retrospect, PEP 673 (which introduced the `Self` type) should have specified this IMO. But if you'd prefer to hold off on that change, then your proposed wording seems fine.